src/options.cpp

```diff
@@ -1320,6 +1322,31 @@ void show_options(bool ingame)
             g->init_ui();
             if( ingame ) {
                 g->refresh_all();
+
+                DebugLog( D_INFO, DC_ALL ) << "Loaded tileset: " << OPTIONS["TILES"].getValue();
+
+                tile_loading_report(termap, "Terrain", "");
+                tile_loading_report(furnmap, "Furniture", "");
+                tile_loading_report(Item_factory::Item_factory().get_all_itypes(), "Items", "");
```

Comment:
`Item_factory::Item_factory()` creates a new (temporary) `Item_factory`, which happens to contain no items because no items have been loaded into it. It is equivalent to `Item_factory()`.

This is for example the same as in `tile_loading_report(termap, "Terrain", std::string());` where the `std::string()` creates a new temporary string object which is used as an argument to the called function.

The call to `get_all_itypes()` happens on that empty object and will never return anything (beside the few build in types like the bionic toolset pseudo item).

You want to use the global static instance of that class:
```C++
item_controller->get_all_itypes()
```

Btw There are certain items that don't need a tile because the are not real and mostly only used for the crafting, e.g. "fire", "apparatus", "toolset", look at `Item_factory::init_old`.

Most of those maps also have null items (e.g. `t_null`, `f_null`, a null vehicle part etc.), you might want to filter those out as they are never supposed to appear - a `f_null` is simply no furniture at all, a `t_null` should never happen, each place should have a terrain etc.